### DISCUSSION TITLE
MC (Midnight Commander) on BigSur 11.2 Apple M1 gets immediately killed

### DISCUSSION THREAD

**Discussion by rafo** [Homebrew/discussions#772] [Everyday usage]
OS: macOS Big Sur 11.2
Hardware: MBP 13" Apple M1 16GB

Everytime i try to start mc it got immediately killed.

`mc zsh: killed     mc`

Uninstall, Install, Reinstall, Update, upgrade doesnt help. Yes, i had the problems mentioned here (https://github.com/Homebrew/brew/issues/10275) but everything seems to be fixed after uninstalling/installing/reinstalling (with cleanup) xcode and git. Every other brew app seems to work fine.

`brew doctor
Your system is ready to brew.`

`brew config
HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d1696e62cc1740dd955566dc79a12ac2fbd70e6c
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.30.1 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
Rosetta 2: false
`

Am I the only one?

**Comment by SMillerDev** [comment]
Is it this? https://github.com/Homebrew/homebrew-core/issues/70326

**Reply by rafo** [reply]
I think its slightly different since my mc doenst start at all. But I admint that it could be mc fault bc I dont find any information on their side about compatibility to Apple silicone or any issues around M1 (only the positive statement on brews own mc side https://formulae.brew.sh/formula/midnight-commander).

**Reply by JimBartlett** [reply]
You are not the only one. The same error: 

zsh: killed mc

Tried under bash, as well as sudo.

**Reply by alexpts** [reply]
I have same problem with zsh in powerlevel10k theme

**Reply by carlocab** [reply]
Try
```
brew reinstall $(brew deps midnight-commander) midnight-commander
```

**Comment by alex-vas** [comment]
Fixed with the following command:

```sh
brew reinstall gettext
```

How I get there:

Checked `Console` (standard macOS application to access logs) under 'Crash Reports', process name 'mc'. It said `Exception Type:        EXC_BAD_ACCESS (Code Signature Invalid)` and further down: `Application Specific Information: dyld: launch, loading dependent libraries /opt/homebrew/opt/gettext/lib/libintl.8.dylib`. So reinstalling `gettext` with `brew reinstall gettext` actually helped.

**Reply by rafo** [reply]
I will test it today.

**Reply by rafo** [reply]
That was not enough for me (see below)

**Comment by AlexanderTyshchenko** [comment]
I have similar problem:

launch Console.app and see crash report

_Application Specific Information:
dyld: launch, loading dependent libraries
/opt/homebrew/opt/pcre/lib/libpcre.1.dylib_



helped
_brew reinstall pcre_

**Reply by rafo** [reply]
That was not enough for me (see below)

**Comment by rafo** [comment]
Thanks for the tip with the crash report. It crashed after every reinstall (below) with a new crashreport leading to next thing I had to reinstall. At the end I had to update:

```
brew reinstall pcre
brew reinstall s-lang
brew reinstall libssh2
brew reinstall git
```

**Reply by JimBartlett** [reply]
I have tried all the reinstalls, but still no luck, same errors.

**Reply by carlocab** [reply]
Try
```
brew reinstall $(brew deps midnight-commander)
```

**Reply by JimBartlett** [reply]
Thank you very much; that did the trick. All is working now.
